src/npc_id.h

```diff
@@ -347,6 +347,7 @@ enum NPCID
 
     NPCID_PEACHBOMB = 291,
     NPCID_BOOMERANG = 292,
+    NPCID_STARMAN = 293,
```

Comment:
Rather than have a fixed starman ID, please add the global temporary `npc_staramn_id` and the NPC.txt option "isstarman" that turns NPC into starman. And at these conditions, use that global variable instead of the constant ID. Also, assign it with -1 by default and in the condition, when it's -1, just make it don't work. So, example condition:
```cpp
if(npc_starman_id > 0 && npc.id == npc_starman_id)
{
    // ...
}
```

Note: global assets support npc.txt, so, you can put at the `graphics/npc/npc-293.txt` fill this field, and make it work.

You'll ask why? In the future, we plan to implement the kind of meta-NPCs, because we want to allow different asset packs to have a different set of incompatible NPCs that don't match between SMBX-38A and SMBX2. In addition, NPCs between 293 and 300 may be used for different purposes in existing episodes.

And we solve two tasks:
- we ensure compatibility with old stuff
- we allow brand new features utilised in new projects